### DISCUSSION TITLE
Overlay Page

### DISCUSSION THREAD

**Discussion by Hackmodford** [xamarin/XamarinCommunityToolkit#1801] [Ideas]
I recently wanted to creating my own Progress HUD.

This can be accomplished by using a modal view with a transparent background. However, this hijacks navigation meaning you cannot navigate to different pages in the background.

I came up with a solution whereby I can display a page as an overlay in iOS/Android. You can still perform regular navigation in the background. This feature could be used to create custom alerts, tutorial overlays, progress indicators, etc.

This seems like a great feature that could be added to XCT.

**Comment by pictos** [comment]
@Hackmodford I would say that our Popup can do this, can you confirm?

**Reply by Hackmodford** [reply]
I don't believe I could get the popup to cover the entire screen or to have a transparent background.

**Reply by pictos** [reply]
Do you have a sample repro on what you did or what you want to archive?

**Reply by Hackmodford** [reply]
I can make one. I just wasn't sure if new features were being added at this time?

**Reply by Hackmodford** [reply]
Here's a demo video of the concept.

Steps:
* present modal login page
* present the overlay page
  * close the modal login page
  * navigate to another view
* close overlay page

The key being I can do all of the navigation behind the overlay. 

![example_overlay](https://user-images.githubusercontent.com/326668/151597590-ac631eaf-df78-47d0-9542-6a83096e9a44.gif)

This is working for me in iOS and Android

**Reply by pictos** [reply]
I can't see how this is different from our popup control

**Reply by pictos** [reply]
> I can make one. I just wasn't sure if new features were being added at this time?

all the new features should be added to the CommunityToolkit.Maui
